test/e2e/resize_nodes.go

```diff
@@ -184,6 +184,24 @@ func waitForPodsCreated(c *client.Client, ns, name string, replicas int) (*api.P
 	return nil, fmt.Errorf("controller %s: Gave up waiting for %d pods to come up", name, replicas)
 }
 
+func waitForPodOnNode(c *client.Client, ns, name, node string) (*api.Pod, error) {
```

Comment:
We already have utility functions for this, as most e2e tests need it.  Please use one of the existing utilities.
As an aside, if you need to retry as below, please use wait.Poll() rather than a home-grown retry loop.